### ISSUE TITLE
bug: `file-management/reference-filenames` neglects keywords

### ISSUE THREAD

**Issue by mainej** [clojure-lsp/clojure-lsp#1018]
**Describe the bug**

`file-management/reference-filenames` doesn't try to detect changes to the usages of registered keywords, and therefore doesn't notify the files where they are registered. Therefore, their unused-public-var lint gets out of date.

Fixing this could cause a serious performance degradation, since so many changes add or remove keywords, and keywords can be registered in any namespace. However, see #1019.

**Comment by mainej** [comment]
> making clojure-lsp analysis differ from kondo a little bit more

That's a valid concern... it's hard to keep track of how our data differs from the [clj-kondo documentation](https://github.com/clj-kondo/clj-kondo/tree/master/analysis). What I find particularly difficult to remember is which keys we've renamed. I assume those renames make  `shared/inside?` and other similar tools a little more generic. But if it were possible to do less of that, it'd be easier to read the clj-kondo docs and trust that it applied to our elements too.

Moving elements around to different buckets is also a bit confusing, but I've memorized that `:namespace-alias` is our custom bucket, and is mostly a copy of `:namespace-usages`. (I've actually thought about proposing that we get rid of `:namespace-alias`, and instead re-calculate whether a `:namepsace-usage` is an alias and what the alias's range is whenever we need it.)

Anyway, I agree, separating keyword definitions will be another thing to remember. :/
